### PR TITLE
Add and use TransactionOptionsDto

### LINKED ISSUE #1692: TransactionOptions should use TaggedDataPayloadDto in message interface
## Bug description

Currently TaggedDataPayload is required, which would look like `"taggedDataPayload":{"tag":{"inner":[1],"bounded":null},"data":{"inner":[1],"bounded":null}}` (tag and data both being [1])

## Expected behaviour

The dto should be used in the message interface (a TransactionOptionsDto should be added I guess), so the bindings can provide it in this format
`"taggedDataPayload": {
            "type": 5,
            "tag": "0x01",
            "data": "0x01"
          }`

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add and use TransactionOptionsDto so the types are consistent

## Links to any relevant issues

Fixes #1692 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/message_interface/message_handler.rs [hunk 1]
```diff
@@ -535,7 +539,17 @@ impl WalletMessageHandler {
             }
             AccountMethod::BurnNft { nft_id, options } => {
                 convert_async_panics(|| async {
-                    let transaction = account_handle.burn_nft(NftId::try_from(&nft_id)?, options).await?;
+                    let transaction = account_handle
+                        .burn_nft(
+                            NftId::try_from(&nft_id)?,
+                            match options {
```
Comment (thibault-martinez) [inline src/message_interface/message_handler.rs]: For this and all of them, looks like you could just map instead of match ?

Comment (Thoralf-M) [reply src/message_interface/message_handler.rs]: But with the map I can't do the error handling? I think there was some way to do the same with less code, but couldn't find it

Comment (thibault-martinez) [reply src/message_interface/message_handler.rs]: Ah yeah right I missed the `?`. Maybe just `transpose()?` after the `map` ?

Comment (Thoralf-M) [reply src/message_interface/message_handler.rs]: Yes that was what I searched, thanks!

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]
Today i learned about `transpose()` 🤯

**Pull request merged**